.github/workflows/ansible-release.yml

```
@@ -49,10 +47,39 @@ jobs:
       - name: Install dependencies
         working-directory: antsibull
         run: |
+          python3 -m pip install packaging
           python3 -m pip install ansible-core
           python3 -m pip install antsibull
           ansible-galaxy install -r requirements.yml
 
+      - name: Validate version and extract major version
+        shell: python
+        id: extract-version
+        run: |
+          import os
+          import pathlib
+          import sys
+
+          from packaging.version import Version
+
+          FILE_APPEND_MODE = 'a'
+          OUTPUTS_FILE_PATH = pathlib.Path(os.environ['GITHUB_OUTPUT'])
+          VERSION = os.environ['ANSIBLE_VERSION']
+
+          def set_output(name, value):
+              with OUTPUTS_FILE_PATH.open(FILE_APPEND_MODE) as outputs_file:
+                  outputs_file.writelines(f'{name}={value}{os.linesep}')
+
+          try:
+              version = Version(VERSION)
+          except Exception as exc:
+              print(
+                  f'::error ::The version {VERSION!r} cannot be parsed: {exc}.'
+              )
+              sys.exit(1)
```

Can be condensed like this if you prefer:

```suggestion
              sys.exit(
                  f'::error ::The version {VERSION!r} cannot be parsed: {exc}.'
              )
```

---

That prints to stderr and not to stdout. Does GHA's `::error ::...` syntax also work with stderr?

---

No idea. Probably better to leave it as.

/resolve

---

> Does GHA's ::error ::... syntax also work with stderr?

Yes, it works, I've done this in the past.